scripts/languageserver/provider_misc.jl

```diff
@@ -75,10 +75,36 @@ end
 abstract didChange <: Method
 
 function Respond(r::Request{didChange,DidChangeTextDocumentParams})
+        vtdi = r.params.textDocument
+        C = r.params.contentChanges
+        for c in C
+            c(vtdi)
+        end
     try
-        documents[r.params.textDocument.uri] = split(r.params.contentChanges[1].text, r"\r\n?|\n")
         return
     catch err
         return Response{didChange,Exception}("2.0",r.id,err)
     end
 end
+
+function (c::TextDocumentContentChangeEvent)(vtdi::VersionedTextDocumentIdentifier)
+    d = join(documents[vtdi.uri],'\n').data
```

Comment:
``Vector{UInt}`` would probably be tricky, then we would have to do the whole UTF-8 handling ourselves... Do we know whether this part is actually a real bottleneck? I would be kind of surprised. Right now even the full sync performs pretty well on my system, so I would think if we were to move to a ``Vector{AbstractString}`` thing, it could only get better.